third-party/build.gradle

```diff
@@ -0,0 +1,193 @@
+    api(":parser") //custom collection of parser packages. Developed by DWR.
+    api(":sqljdbc4-2.0") //matching jar unavilable in maven central. Newer version available
+    api(":swixml") //local jar contains additional classes compared to 1.5.144 jar on maven central
+    api(":vista") //developed by DWR. Unavailable on maven central
+    api(":XAOptimizer") //Unavailable on maven central
+    api(":xml") //custom collection of sun xml packages
+
+    //dlls
+    api fileTree(dir: '../libs/x64', include: ['cb.dll'])
+    api fileTree(dir: '../libs/x64', include: ['jep.dll'])
+
+    //sys jars
+    api(":codebase")
+    api(":jai_codec")
+    api(":jai_core")
+    api(":jai_imageio")
+    api(":jdom")
+    api(":jh")
+    api(":jxl")
+    api(":jython")
+}
+
+configurations.configureEach {
+    exclude group: "javax.media", module: "jai_core"
+    exclude group: "javax.media", module: "jai_codec"
+    exclude group: "com.sun.media", module: "jai_imageio"
+}
+
+tasks.register('copyDependencies', Copy) {
+    from(configurations.runtimeClasspath)
+    into 'build/tmp/libs'
+    duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+}
+
+jar.dependsOn(copyDependencies)
+
+tasks.register('prepareBndFile') {
+    def bndFile = file('bnd.bnd')
+    def bndTemplateFile = file('bnd.bnd.template')
+    def libsDir = file('build/tmp/libs')
+    def packageSet = new LinkedHashSet<String>()
+    def includeResourceSet = new LinkedHashSet<String>()
+
+    doFirst {
+        println "Running prepareBndFile task..."
+
+        // Delete bnd.bnd if it exists
+        if (bndFile.exists()) {
+            bndFile.delete()
+            println "Deleted existing bnd.bnd file."
+        }
+
+        // print the full path to the bndFile
+        println "bndFile: ${bndFile}"
+
+        // Copy bnd.bnd.template to bnd.bnd
+        bndFile.text = bndTemplateFile.text
+        println "Copied bnd.bnd.template to bnd.bnd."
+
+        // Collect package names from JAR files
+        libsDir.eachFileRecurse { file ->
+            if (file.name.endsWith('.jar')) {
+                includeResourceSet.add("lib_x64/${file.name}=${file.path.replace('\\', '/')}")
+                JarFile jar = new JarFile(file)
+                jar.entries().each { entry ->
+                    if (entry.name.endsWith('.class')) {
+                        def packageName = entry.name.replaceAll('/[^/]+$', '').replace('/', '.')
+                        if (!packageName.startsWith('Lib.')) {
+                            packageSet.add(packageName)
+                        }
+                    }
+                }
+            }
+        }
+
+        // Add Export-Package contents to the end of bnd.bnd
+        def exportContents = packageSet.collect { it + ',\\' }.join('\n    ')
+        bndFile.append("\nExport-Package: \\\n    ${exportContents}\n")
+        println "Added Export-Package contents to bnd.bnd."
+    }
+}
+
+jar.dependsOn(prepareBndFile)
```

Comment:
The behavior here is very different than the previous plugin behaviors. When I test this out locally, I'm seeing a jar with all of the class files put into a fat jar, rather than seeing the directory of jars inside my jar like I expected.


<img width="688" alt="image" src="https://github.com/user-attachments/assets/3025bb36-062e-48f1-9670-02e5588a256a">


compared to:
![image](https://github.com/user-attachments/assets/0f08bc97-87e2-4dfa-ad2c-e9e32397886c)


Is this a limitation of the plugin?

Reply:
As far as I have been able to determine, this is a limitation of the way that the third-party bnd bundle is being dynamically generated. If this proves to be an issue than I will find a way to bundle the third party dependencies as full jars.

Reply:
is this documented somewhere for when deliverable comparisons are performed?

Reply:
Stephen was able to help identify an alternative way to declare the exports that doesn't automatically unpack them using "-exportcontents" instead of "Export-Package" in the bnd configuration. Third party is now bundling the jars like hecjars.